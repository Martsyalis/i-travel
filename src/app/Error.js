import React from 'react';

export default ({ error }) => {
  if(!error) return null;
  return (
    <div className="error">
      {Array.isArray(error) 
        ? <ul>error.map(err => <li>err</li>)</ul>
        : error.error ? error.error : error
      }
    </div>
  );
};