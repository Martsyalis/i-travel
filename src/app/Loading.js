import React from 'react';

export default ({ loading }) => {
  if (!loading) return null;
  return <div className="loader">Loading...</div>;
};
