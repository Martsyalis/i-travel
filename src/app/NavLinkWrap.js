import React, { PureComponent } from 'react';
import { NavLink, withRouter } from 'react-router-dom';

class NavLinkWrap extends PureComponent {
  render() {
    return (
      <NavLink
        to={this.props.to}
        activeClassName="active"
        className="navbar-item"
      >
        {this.props.to === '/' ? (
          <img src={require('../resources/logo.png')} alt="Travel_Share Logo" />
        ) : (
          this.props.to.charAt(1).toUpperCase() + this.props.to.slice(2)
        )}
      </NavLink>
    );
  }
}

export default withRouter(NavLinkWrap);
