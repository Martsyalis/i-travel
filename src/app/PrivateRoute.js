import React, { PureComponent } from 'react';
import { Route, Redirect } from 'react-router-dom';
import { connect } from 'react-redux';

class PrivateRoute extends PureComponent {
  render() {
    const { Component, render, user, ...rest } = this.props;
    return(
      <Route {...rest} render={props => {
        if(user) {
          console.log('user is with render', user, render);
          return render ? render(props) : <Component {...props}/>;
        }
        console.log('we sholdnt be here');
        return (
          <Redirect to={{
            pathname: '/auth/signin',
            state: { from: props.location.pathname }
          }}/>
        );
      }}/>
    );
  }
}

export default connect(
  state => ({ user: state.auth.user }),
  null,
  (stateProps, dispatchProps, ownProps) => ({ 
    user: stateProps.user,
    Component: ownProps.component
  })
)(PrivateRoute);