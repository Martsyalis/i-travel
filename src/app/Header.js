import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import  NavLinkWrap from './NavLinkWrap';
import { NavLink, withRouter } from 'react-router-dom';
import { signout } from '../auth/actions';

class Header extends PureComponent {
  render() {
    return (
      <div className="header">
        <nav className="breadcrumb">
            <NavLinkWrap to="/home"/>
            <NavLinkWrap to="/search"/>            
            <NavLinkWrap to="/upload"/>
            <NavLinkWrap to="/shared"/>
            {this.props.user 
              ? <NavLink to="/" onClick={this.props.signout}> Sign Out </NavLink>
              : <NavLink to="/auth/signin"> Sign In </NavLink>
            }
        </nav>
      </div>
    );
  }
}

export default withRouter(connect(state => ({ user: state.auth.user }), { signout }) (Header));