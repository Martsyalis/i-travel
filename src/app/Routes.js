import React from 'react';
import {
  Route,
  Switch,
  Redirect
} from 'react-router-dom';
import Home from '../home/Home';
import Auth from '../auth/Auth';
import PrivateRoute from './PrivateRoute';
import Experience from '../experience/Experience';


export default () => (
  <Switch> 
    <Route path="/auth" render={() => <Auth/>}/>;
    <PrivateRoute exact path="/" component={Home}/>;
    {/* <PrivateRoute path="/experience/:id" component={Experience}/> */}
    <PrivateRoute path="/experience/:id" render={({ match }) => <Experience id={match.params.id}/>}/>
    <Redirect to="/"/>
  </Switch>  
);
