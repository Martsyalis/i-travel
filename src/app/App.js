import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import './App.css';
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Redirect
} from 'react-router-dom';
import Header from './Header'; 
import Error from './Error';
import Loading from './Loading';
import Home from '../home/Home';
import Search from '../search/Search';
import UploadForm from '../upload/UploadForm';

class App extends PureComponent {
  
  render() {
    const { loading, error } = this.props;

    return (
      <Router>
        <div className="App">
          <Header/>
          <main>
            <Switch>
              <Route exact path="/" component = {Home}/>
              {/* <Route exact path="/search" component = {Search}/>
              <Route exact path="/upload" component = {UploadForm}/> */}
              <Redirect to="/"/>
            </Switch>
            <Loading loading={loading}/>
            <Error error={error}/>
          </main>
        </div>
      </Router>
    );
  }
}

export default connect(
  state => ({ 
    loading: state.loading,
    error: state.error
  }),
  null
)(App);
