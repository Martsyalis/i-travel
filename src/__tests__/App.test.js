import React from "react";

import { shallow } from "enzyme";

import { App } from "../app/App";

describe("App", () => {
  it("shows the App component", () => {
    const checkedToken = "test";
    const loading = false;
    const error = null;
    const checkForToken = () => {};

    const wrapper = shallow(
      <App
        checkedToken={checkedToken}
        loading={loading}
        error={error}
        checkForToken={checkForToken}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
