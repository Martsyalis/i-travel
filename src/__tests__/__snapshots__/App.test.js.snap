// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App shows the App component 1`] = `
<BrowserRouter>
  <div>
    <div
      className="App"
    >
      <Connect(Header) />
      <main>
        <Component />
      </main>
    </div>
  </div>
</BrowserRouter>
`;
